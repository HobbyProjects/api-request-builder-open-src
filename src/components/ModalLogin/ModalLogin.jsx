// Copyright DocuSign, Inc. â’¸ 2020. MIT License -- https://opensource.org/licenses/MIT
import React from 'react';
import { ModalFace, ButtonFace } from '../../UxFaces';
//import { thisTypeAnnotation } from '@babel/types';

class ModalLogin extends React.Component {
    constructor(props) {
        // prop appObject is required //
        super(props);
        this.state = {visible: false, msg: null};
        this.close = this.close.bind(this);
        this.loginActionHandler = this.loginActionHandler.bind(this);
    }

    show (msg) {
        this.props.appObject.blocklySection.blocklyClearSelection();
        this.setState({visible: true, msg: msg})
    }        
    close () {this.setState({visible: false})}        

    render () {
        return (
            <ModalFace
                {...this.props}
                visible={this.state.visible}
                closeButton={<ModalFace.Close onClick={this.loginActionHandler}/>}
                onClose={this.loginActionHandler}
            >
                <ModalFace.Header title="Please login"/>
                <ModalFace.Body>
                    <div>
                        {this.state.msg ? 
                        <p>{this.state.msg}</p> :
                        <>
                        <p>Please login using your developer demo account.</p>
                        <p className="topmargin">Don't have a developer demo account? Get 
                            a <a rel="noopener noreferrer" target="_blank" href="https://go.docusign.com/o/sandbox/">free account!</a></p>
                        </>
                        }
                    </div>
                </ModalFace.Body>
                <ModalFace.Footer>
                    <ButtonFace
                        kind="primary"
                        onClick={this.loginActionHandler}
                        size="large"
                        text="Login"
                    />
                </ModalFace.Footer>
            </ModalFace>
        )
    }

    loginActionHandler() {this.close(); this.props.dsAuth.startLogin()}
}

export default ModalLogin;
