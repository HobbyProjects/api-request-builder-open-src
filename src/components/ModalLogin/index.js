// Copyright DocuSign, Inc. â’¸ 2020. MIT License -- https://opensource.org/licenses/MIT
import ModalLogin from './ModalLogin';
export default ModalLogin;

