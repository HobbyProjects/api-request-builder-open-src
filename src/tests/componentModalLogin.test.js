// Copyright DocuSign, Inc. â’¸ 2020. MIT License -- https://opensource.org/licenses/MIT
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { Olive, Modal } from '@olive/react'

import ModalLogin from '../components/ModalLogin';
Enzyme.configure({ adapter: new Adapter() })


it('ModalLogin renders closed', () => {
  const tree = renderer
    .create(
        <Olive><Modal.Portal><ModalLogin/></Modal.Portal></Olive>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ModalLogin opens', () => {
    const appObject = {blocklySection: {blocklyClearSelection: () => {}}}
    const wrapper = shallow(<ModalLogin appObject={appObject}/>);
    const msg = "THIS IS THE MESSAGE";
    wrapper.instance().show(msg);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.contains(<p>{msg}</p>)).toBe(true);
});