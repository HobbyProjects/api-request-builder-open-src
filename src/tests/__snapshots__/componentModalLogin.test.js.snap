// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLogin opens 1`] = `
<Modal
  appObject={
    Object {
      "blocklySection": Object {
        "blocklyClearSelection": [Function],
      },
    }
  }
  closeButton={
    <Modal.Close
      onClick={[Function]}
    />
  }
  height="content"
  onClose={[Function]}
  shrouded={false}
  visible={true}
  width="medium"
>
  <Modal.Header
    title="Please login"
  />
  <Modal.Body
    noPadding={false}
  >
    <div>
      <p>
        THIS IS THE MESSAGE
      </p>
    </div>
  </Modal.Body>
  <Modal.Footer>
    <Button
      accessibilityText=""
      active={false}
      disabled={false}
      fullWidth={false}
      hideTrigger={false}
      href=""
      iconPosition="beforeText"
      inverted={false}
      kind="primary"
      menuTrigger={false}
      onClick={[Function]}
      size="large"
      text="login"
      type="button"
    />
    <Button
      accessibilityText=""
      active={false}
      disabled={false}
      fullWidth={false}
      hideTrigger={false}
      href=""
      iconPosition="beforeText"
      inverted={false}
      kind="tertiary"
      menuTrigger={false}
      onClick={[Function]}
      size="large"
      text="Cancel"
      type="button"
    />
  </Modal.Footer>
</Modal>
`;

exports[`ModalLogin renders closed 1`] = `
Array [
  <div
    aria-live="polite"
    aria-relevant="additions"
    id="SkipNavContainer"
  />,
  <div
    aria-live="polite"
    aria-relevant="additions"
    className="css-cgizjj-Olive"
    id="ToastMessageContainer"
    role="status"
  />,
  <div
    id="ModalContainer"
  />,
  <div
    aria-live="polite"
    aria-relevant="additions"
    id="OverlayContainer"
    role="dialog"
  />,
]
`;
